src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -284,6 +298,24 @@ public void Dispose()
             Dispose(true);
         }
 
+        internal bool IsEnoughMemory(long hostProcessSize, IEnumerable<long> workerChannelSizes, long memoryLimit)
+        {
+            if (memoryLimit <= 0)
+            {
+                return true;
+            }
+
+            // Checking memory before adding a new worker
+            long maxWorkerSize = workerChannelSizes.Max();
+            long currentMemoryConsumption = workerChannelSizes.Sum() + hostProcessSize;
+            if (currentMemoryConsumption + maxWorkerSize > memoryLimit * 0.8)
```

Comment:
Should we document the reasoning behind this logic here? I do not think I understood why we are adding the `maxWorkerSize` in the expression in line 311

Reply:
By adding `maxWorkerSize` we are predicting what will be overall consumption after adding a new worker. I added a  comment to clarify that.